### ISSUE TITLE
Transient issue on 0.8.1: `duckdb.TransactionException: TransactionContext Error: Failed to commit: Input is invalid/unsupported GZIP stream`

### ISSUE THREAD

**Issue by b-per** [duckdb/duckdb#8420]
### What happens?

I run a pipeline to ingest multiple `json.gz` files daily into a DuckDB database, using a code like 
```
insert into my_table by name select 
            * 
        from read_json_auto('/folder/my_file.json.gz', sample_size=10000)
```

This used to work without a problem with DuckDB 0.7 but since I upgrade to 0.8.1 recently, I get transient errors of 
```
duckdb.TransactionException: TransactionContext Error: Failed to commit: Input is invalid/unsupported GZIP stream
```

Re-running my pipeline a second time usually fixes the problem though

### To Reproduce

I can't reproduce the issue as it comes and goes.

### OS:

x86_64

### DuckDB Version:

0.8.1

### DuckDB Client:

Python

### Full Name:

Benoit Perigaud

### Affiliation:

NA

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by b-per** [comment]
I can actually reproduce the issue in 0.8.1 every time with the following code.
It usually dies at the iteration 2 or 3 in my loop

```
import duckdb

conn = duckdb.connect("mydb.db")
conn.sql("create or replace table test_table as select * from read_json_auto('myfile.json.gz', sample_size=10000)")
for i in range(10):
    print(i)
    conn.sql("insert into test_table select * from read_json_auto('myfile.json.gz', sample_size=10000)")
```

I just tried with `0.8.2.dev2320` now and ran it a few times without a problem. The issue now is that the DB format is different between `0.8.1` and `0.8.2.dev` so I'll need to update the CLI as well and export/import my DB to the new format.

**Comment by b-per** [comment]
And finally, I also tried in `0.8.1` to un-gzip the file before loading it but I still get the same error despite the file being a json one and not a gzip.
```
duckdb.Error: Invalid Error: Input is invalid/unsupported GZIP stream
```

**Comment by b-per** [comment]
I have also tried closing the connection after each step and reopening for the next iteration of the loop and in that case there is no error in `0.8.1`

**Comment by begelundmuller** [comment]
Hello,

At Rill Data, we have also seen this issue intermittently since v0.8, but have not been able to replicate it. Here are some additional findings that can hopefully help you triage the issue:

- The problem usually arises after a connection has been open and used for a while (rarely on a first run).
- We have seen the issue when loading files that have nothing to do with GZIP, such as JSON and Parquet files. The queries we run take the form `INSERT INTO x BY NAME (SELECT * FROM read_xxx(...))`.
- We have also seen the issue when loading data using the DuckDB appender C API
  - In that instance, we got this internal error on the next DuckDB query: `INTERNAL Error: Could not find node in column segment tree!\nAttempting to find row number \"0\" in 0 nodes\n`

Overall, the error does not feel related to GZIP input. I noticed that DuckDB also uses GZIP in its internal storage format for overflow strings. Could some corruption be happening there?

**Comment by hannes** [comment]
If it's intermittent is it maybe possible to share a reproduction script that when ran enough times eventually shows the issue? Because that would already help a lot tracking this down.

**Comment by lostmygithubaccount** [comment]
I keep getting it occasionally in CI/CD that runs every 3 hours, see the export to MotherDuck step: https://github.com/lostmygithubaccount/ibis-analytics/actions/runs/5922595390/job/16056805728

**Comment by hannes** [comment]
@b-per can you share the file that demonstrates the issue please?

**Comment by Stongtong** [comment]
I use 0.7.1 duckdb and there also exists this problem, I use java call duckdb ,
`### Error querying database.  Cause: java.sql.SQLException: Invalid Error: Input is invalid/unsupported GZIP stream\\\\n
### The error may exist in xxxxxMapper.java (best guess)\\\\n
### The error may involve xxxxxMapper.selectWithTimeInsensitive-Inline\\\\n
### The error occurred while setting parameters\\\\n
### SQL: select * from XXXX where namespace ilike ? and name ilike ? and params ilike ? order by update_time desc limit 1\\\\n
### Cause: java.sql.SQLException: Invalid Error: Input is invalid/unsupported GZIP stream\\\\n; uncategorized SQLException; SQL state [null]; error code [0];
 Invalid Error: Input is invalid/unsupported GZIP stream; nested exception is java.sql.SQLException: Invalid Error: Input is invalid/unsupported GZIP stream`

**Comment by Mause** [comment]
@Stongtong do you see the issue in 0.8.1? And can you provide the file?

**Comment by Stongtong** [comment]
@Mause I not upgrade the version to 0.8.1 yet ,the db file is 12GB and share it difficultly

**Comment by nicku33** [comment]
We have also seen this at MotherDuck in production. Have attempted to reproduce with no luck
 * repeatedly loading from parquet
 * repeatedly loading from a dataframe
 * same but with large strings to trigger Overflow paegs
 * in a fuzz test where SELECTS, UPDATES, FORCE CHECKPOINT and loading CTAS activity done over 8 threads, resetting connection at times

Still trying to gather more information. However we ran into an issue yesterday where within 15 min of GZIP errors, we got the column segment tree issue from #3789 that corrupted database. I'm not certain which is cause and which is effect. I suspect that WAL replay during checkpoints may be playing a role here. The exception happens because a block that is supposed to have the 4 byte magic string to signify GZIP is missing them.

Agree with @begelundmuller that this doesn't seem related to GZIP input, but rather internal GZIP compression atm.

**Comment by Alex-Monahan** [comment]
In case it is helpful, another user encountered this issue when inserting into MotherDuck in this GitHub action (in the "export to MotherDuck" section): 
https://github.com/lostmygithubaccount/ibis-analytics/actions/runs/6039101657/job/16386985693

**Comment by Mytherin** [comment]
This should be solved now in #8943 - would be great if you could verify that the problem is solved in the latest main

**Comment by bjornasm** [comment]
I just encountered this problem running in the latest version of duckdb (0.10). Happens when I try to drop a table. @Mytherin
